legacy/reqcounter/reqcounter_test.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reqcounter_test
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	rc "sigs.k8s.io/k8s-container-image-promoter/legacy/reqcounter"
+)
+
+func TestInit(t *testing.T) {
+	rc.Init()
+	// Ensure request counting is enabled.
+	if rc.EnableCounting == false {
```

Comment:
Looks like you're doing a number of `if-statement` tests here to check various properties. The more succinct way to do it is to create an `expectedRc` variable of type `RequestCounter` and to just use `require.Equal(t, expected, actual)`. That way (1) you don't have to write custom error messages for every single possible failure and (2) the testing code can be much shorter and simpler, because all of the properties can be captured in an "expected" RequestCounter (or some other struct) value.

Keep the above in mind for future PRs.

Reply:
Noted. I have reformatted some of the tests to follow this pattern.